### ISSUE TITLE
how to add a variable with dependent indexes in a MIPmodel ?

### ISSUE THREAD

**Issue by gguernec** [dirkschumacher/ompr#320]
Hello,

I would like to create a MIPmodel using ompr. One of the variable to add must have the following structure:

add_variable(reg_absA[x1,x2,x3], x1 = 1:5, x2 = voisins[[x1]], x3= 1:4, type = "continuous", lb = 0) %>%

where "voisin" is a list object of indexes previously defined. When I run the script, the answer is that "x1 is not found". The problem is linked to the expression voisins[[x1]] and more particularly to the fact that **x2 must depend on x1**. Moreover if I replace x2 = voisins[[x1]] by x2 =1:3, everything works fine.

Have you any idea to solve this problem using ompr ?

Thanks by advance

**Comment by gguernec** [comment]
does this associated error message inspire you ?

Error in add_constraint_(.model, lazyeval::as.lazy(substitute(.constraint_expr),  :   object 'reg_absA' not found

**Comment by snestler** [comment]
I have a similar problem with add_variable when trying to use a dependent index.  Any resolution or suggestions here?  Thank you.

**Comment by sbmack** [comment]
> 
> 
> I have a similar problem with add_variable when trying to use a dependent index. Any resolution or suggestions here? Thank you.

I've posted similar questions.  It appears that multi-indexes can only be generated analogous to the `expand.grid` function in base R.  i.e. exhaustive combinations.  With only simple filtering allowed e.g. `i !=  j`  There is no apparent way from what I have seen to define a sparse matrix otherwise.

I hope Dirk can get around to adding that functionality.

**Comment by dirkschumacher** [comment]
I think this might work with the latest `master` version and the `MIPModel`.

**Comment by dirkschumacher** [comment]
I close the issue as it should work with the current `MIPModel` (not on CRAN yet as of writing). If the problem still persists, then it is a bug and please open another issue.

Thanks for reporting that!
